pkg/util/workqueue/delaying_queue_test.go

```diff
@@ -0,0 +1,65 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package workqueue
+
+import (
+	"reflect"
+	"testing"
+	"time"
+)
+
+func TestSimpleQueue(t *testing.T) {
+	q := NewDelayingQueue()
+
+	first := "foo"
+
+	q.AddAfter(first, 50*time.Millisecond)
```

Comment:
This has the potential to be the source of a lot of test flakes. Can you modify the main code to use `util.Clock` with `RealClock` and have the tests use `FakeClock` so you can control it?